src/foundry/foundry.js/hooks.d.ts

```diff
@@ -1,830 +1,804 @@
+     * PlaceableObject.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link PlaceablesLayer#pasteObjects}
      */
-    renderChatMessage: (message: ChatMessage, jq: JQuery, messageData: ChatMessage.MessageData) => unknown;
+    type PastePlaceableObject<P extends PlaceableObject = PlaceableObject> = (
+      copied: P[],
+      pasted: P[]
+    ) => boolean | void;
 
     /**
-     * This is called after {@link AVSettings} are changed.
-     * @param settings - the AVSettings
-     * @param changed  - an object reflecting the changed settings
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link AVSettings#_onSettingsChanged}
+     * @remarks This is called after copying {@link Wall}s in a copy-paste action, but before embedding them into the
+     * {@link WallsLayer}.
+     * @param copied - the originally copied Walls
+     * @param pasted - the pasted copies with new coordiantes
+     * @param W      - the type of the Wall
+     * @remarks The name for this hook is dynamically created by joining 'paste' with the type name of the Wall.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link WallsLayer#pasteObjects}
      */
-    rtcSettingsChanged: (settings: DeepPartial<AVSettings.Settings>, changed: object) => unknown;
+    type PasteWall<W extends Wall = Wall> = (copied: W[], pasted: W[]) => boolean | void;
 
     /**
-     * This is called before the {@link Game} is fully set up.
-     * @remarks This is called by {@link Hooks.callAll}.
-     * @see {@link Game#setupGame}
+     * A hook event that fires for every Document type before execution of a creation workflow. Substitute the
+     * Document name in the hook event to target a specific Document type, for example "preCreateActor". This hook
+     * only fires for the client who is initiating the creation request.
+     *
+     * The hook provides the pending document instance which will be used for the Document creation. Hooked functions
+     * may modify that data or prevent the workflow entirely by explicitly returning false.
+     *
+     * @param document - The pending document which is requested for creation
+     * @param data     - The initial data object provided to the document creation request
+     * @param options  - Additional options which modify the creation request
+     * @param userId   - The ID of the requesting user, always game.user.id
+     * @returns Explicitly return false to prevent creation of this Document
+     * @remarks The name for this hook is dynamically created by joining 'preCreate' with the name of the Document.
+     * @remarks This is called by {@link Hooks.call}.
+     * @see {@link ClientDatabaseBackend#_preCreateDocumentArray}
+     * @see {@link TokenDocument#_preUpdateTokenActor}
      */
-    setup: () => unknown;
+    type PreCreateDocument<D extends DocumentConstructor = DocumentConstructor> = (
+      document: D,
+      data: ConstructorParameters<D>,
```

Comment:
That's not correct, it's only the first parameter, not all of them. Also might be better to get this explicitly with the `_initializeSource` trick (or create a helper type for that).